docs/docs/reference/config-files/gatsby-config.md

```diff
@@ -161,7 +164,17 @@ module.exports = {
 
 See more about [Adding a Path Prefix](/docs/how-to/previews-deploys-hosting/path-prefix/).
 
-## Polyfill
+## trailingSlash
+
+Configures the creation of URLs and whether to remove, append, or ignore trailing slashes.
+
+- `always`: Always add trailing slashes to each URL, e.g. `/x` to `/x/`.
+- `never`: Remove all trailing slashes on each URL, e.g. `/x/` to `/x`.
+- `ignore`: Don't automatically modify the URL
+
+Until Gatsby v4 it'll be set to `legacy` by default, in Gatsby v5 the default mode will be `always`.
```

Comment:
What's the difference between ignore and legacy? Do we need two options there?

Reply:
It's due to File System Route API where currently it always adds a trailing slash:

https://github.com/gatsbyjs/gatsby/blob/992c6d66ff59eb976f6a5c5985dc33f2054a97dc/packages/gatsby-page-utils/src/create-path.ts#L20-L25

So, if you have e.g. a markdown file with `slug: /without` in frontmatter (+ `{MarkdownRemark.frontmatter__slug}.js` it'll still output `/without/`. Because of the last `/` in the `posix.join()`. That's the legacy behavior. With manual `createPage` call it respects that.

Reply:
So if a user doesn't set an option the default will be `legacy` and in v5 it'll be `always`? To avoid breaking changes?

Reply:
Correct